sabnzbd/downloader.py

```diff
@@ -465,18 +496,25 @@ def run(self):
                             request_server_info(server)
                         break
 
-                    article = sabnzbd.nzbqueue.NzbQueue.do.get_article(server, self.servers)
+                    if not server.article_queue:
+                        fetch_amount = server.threads * 2
+                        if fetch_amount > 50:
+                            fetch_amount = 50
+                        server.article_queue = sabnzbd.nzbqueue.NzbQueue.do.get_articles(server, self.servers, fetch_amount)
```

Comment:
I don't really know if there is a good number of articles to fetch. The benefit is reduced by fetching too few but 50 may be excessive.